twitchio/ext/routines/__init__.py

```diff
@@ -330,6 +330,29 @@ def start_time(self) -> Optional[datetime.datetime]:
         """
         return self._start_time
 
+    @property
+    def next_run(self) -> Optional[datetime.datetime]:
```

Comment:
I'll test the logic later but this can be simplified in design and remove nesting

```py
    @property
    def next_run(self) -> Optional[datetime.datetime]:
        """Calculate and return the next run time of the routine."""
        if self._time:
            next_run = self._time + datetime.timedelta(days=self._completed_loops)
            now = datetime.datetime.now(datetime.timezone.utc)
            if next_run < now:
                next_run += datetime.timedelta(days=1)
        elif self._delta:
            next_run = self._start_time + datetime.timedelta(seconds=self._delta * (self._completed_loops + 1))
            now = datetime.datetime.now(datetime.timezone.utc)
            if next_run < now:
                next_run = now + datetime.timedelta(seconds=self._delta)
        else:
            return None

        return next_run
```

Reply:
I have updated to remove nesting.